### PR TITLE
Tidy and add tests for atoms/lp_cone/DotSortAtom.jl

### PR DESCRIPTION
**Pull request by odow**

### CODE REVIEW

File: src/atoms/lp_cone/DotSortAtom.jl [hunk 1]
```diff
@@ -32,6 +23,7 @@ function Base.sign(x::DotSortAtom)
     if all(x.w .>= 0)
         return sign(x.children[1])
     elseif all(x.w .<= 0)
+        # FIXME: shouldn't this be -sign?
         return sign(x.children[1])
```
Comment (odow) [inline src/atoms/lp_cone/DotSortAtom.jl:27]: @ericphanson just to double-check my understanding, shouldn't this be `-sign(x.children[1])`?

Comment (odow) [reply src/atoms/lp_cone/DotSortAtom.jl:27]: This goes right back to the original commit: https://github.com/jump-dev/Convex.jl/commit/4795ded2fcf2a15a0ebf85ab7eb49418d5eb67fd

File: test/test_atoms.jl [hunk 2]
```diff
@@ -767,7 +767,50 @@ end
...
+    minobjective: 1.0 * u1 + u2 + v1 + v2
+    [v1 + u1, v1 + u2, v2 + u1 + -1.0*x1, v2 + u2 + -1.0*x2] in Nonnegatives(4)
+    """
+    _test_atom(target) do context
+        return dotsort(Variable(2), [0, 1])
+    end
+    target = """
+    variables: v1, v2, u1, u2, x1, x2
+    minobjective: 1.0 * u1 + u2 + v1 + v2
+    [v1 + u1 + 2.0*x1, v1 + u2 +2.0*x2, v2 + u1 + -3.0*x1, v2 + u2 + -3.0*x2] in Nonnegatives(4)
```
Comment (odow) [inline test/test_atoms.jl:782]:
```suggestion
    [v1+u1+2.0*x1, v1+u2+2.0*x2, v2+u1+-3.0*x1, v2+u2+-3.0*x2] in Nonnegatives(4)
```

File: test/test_atoms.jl [hunk 3]
```diff
@@ -767,7 +767,50 @@ end
...
+    """
+    _test_atom(target) do context
+        return dotsort(Variable(2), [-2, 3])
+    end
+    target = """
+    variables: v1, v2, u1, u2, x1, x2
+    minobjective: 1.0 * u1 + u2 + v1 + v2
+    [v1 + u1 + 2.0*x1, v1 + u2 +2.0*x2, v2 + u1 + 3.0*x1, v2 + u2 + 3.0*x2] in Nonnegatives(4)
```
Comment (odow) [inline test/test_atoms.jl:790]:
```suggestion
    [v1+u1+2.0*x1, v1+u2+2.0*x2, v2+u1+3.0*x1, v2+u2+3.0*x2] in Nonnegatives(4)
```

File: test/test_atoms.jl [hunk 4]
```diff
@@ -767,7 +767,50 @@ end
 
 ### lp_cone/DotSortAtom
 
-# TODO
+function test_DotSortAtom()
+    target = """
+    variables: v1, v2, u1, u2, x1, x2
+    minobjective: 1.0 * u1 + u2 + v1 + v2
+    [v1 + u1, v1 + u2, v2 + u1 + -1.0*x1, v2 + u2 + -1.0*x2] in Nonnegatives(4)
```
Comment (odow) [inline test/test_atoms.jl:774]:
```suggestion
    [v1+u1, v1+u2, v2+u1+-1.0*x1, v2+u2+-1.0*x2] in Nonnegatives(4)
```

File: src/atoms/lp_cone/DotSortAtom.jl (hunk 1 above)
Comment (ericphanson) [reply src/atoms/lp_cone/DotSortAtom.jl:27]: yeah, you are right, looking at the `sign_from_args` part of https://www.cvxpy.org/_modules/cvxpy/atoms/dotsort.html#dotsort.

I never did a thorough sweep of the repo as you're doing, and I'm very glad you're doing it!!

Comment (odow) [reply src/atoms/lp_cone/DotSortAtom.jl:27]: Okay, cool. I didn't think to look at the cvxpy implementation.

> I never did a thorough sweep of the repo as you're doing, and I'm very glad you're doing it!!

I have been learning a lot!

**Pull request merged**